### ISSUE TITLE
Test flake in TestPolicyBasedRestrictionOfBuildCreateAndCloneByStrategy

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#5410]
```
I1026 20:07:52.886170   28086 controller.go:84] Updated all DeploymentConfigs for trigger on ImageStream integration/builderimage
I1026 20:07:52.888302   28086 image_change_controller.go:47] Build image change controller detected ImageStream change 
I1026 20:07:52.987682   28086 controller.go:84] Updated all DeploymentConfigs for trigger on ImageStream integration/builderimage
I1026 20:07:53.006215   28086 helper.go:376] UpdateTrackingTags: stream=integration/builderimage, updatedTag=latest, updatedImage.dockerImageReference=test/builderimage:latest, updatedImage.image=image-id
--- FAIL: TestPolicyBasedRestrictionOfBuildCreateAndCloneByStrategy-16 (7.14s)
	build_admission_test.go:171: Couldn't create ImageStreamMapping: imageStream "builderimage" cannot be updated: the object has been modified; please apply your changes to the latest version and try again
```

**Comment by deads2k** [comment]
@csrwng This test looks like the admission control one for builds.

**Comment by deads2k** [comment]
you could consider using patch instead of replace, it will retry on changes as long as they are non-conflicting changes.

**Comment by csrwng** [comment]
@deads2k the test is not updating the ImageStream directly, it's doing so indirectly by trying to add the imagestreamtag... I guess I can add the retry logic in the test.

**Comment by deads2k** [comment]
> @deads2k the test is not updating the ImageStream directly, it's doing so indirectly by trying to add the imagestreamtag... I guess I can add the retry logic in the test.

Should the `imagestreammapping.REST` be retrying a couple times for you or would that be incorrect?  It seems like you'd be safe to retry as long as the update did not mutate anything related to your tag, right?

**Comment by smarterclayton** [comment]
We added a retry in image stream import - maybe it was in the controller, not in ISM

**Comment by deads2k** [comment]
looks like a dupe of https://github.com/openshift/origin/issues/5421

**Comment by soltysh** [comment]
#5421 was resolved with #5455 please close.
